examples/asr/asr_adapters/train_asr_adapter.py

```diff
@@ -78,8 +78,15 @@ def update_encoder_config_to_support_adapter(model_cfg):
 
 
 def update_model_cfg(original_cfg, new_cfg):
-    with open_dict(new_cfg):
-        # drop keys which dont exist in old config
+    with open_dict(original_cfg), open_dict(new_cfg):
+        # force inject some keys into the config
+        whitelist_keys = ['num_workers', 'pin_memory']
+        for wkey in whitelist_keys:
+            if wkey in new_cfg:
+                original_cfg[wkey] = new_cfg[wkey]
```

Comment:
Is it a good idea to change the original_cfg as it is just an input to the model?
It can be used in other places after calling this method.
Also it assumes that original_cfg has both the num_workers and pin_memory already in itself. Is this assumption always true?

Reply:
Thats why this white list is used - these two keys may be missing from older models, therefore we check if the new config (that is provided to train an adaper) contained these keys, if so then override the old config value (whether or not it had them).

This is necessary to prevent old citrinet and contextnet models from training with 0 workers.